alica_engine/src/engine/model/Plan.cpp

```diff
@@ -106,9 +105,34 @@ void Plan::setTransitions(const TransitionGrp& transitions)
     _transitions = transitions;
 }
 
-void Plan::setBasicPlan(std::unique_ptr<BasicPlan>&& basicPlan)
+void Plan::computeDynamicEntryPoints() const
 {
-    _basicPlan = std::move(basicPlan);
+    _allEntryPoints.clear();
+    for (auto ep : _entryPoints) {
+        if (!ep->isDynamic()) {
+            _allEntryPoints.push_back(ep);
+            continue;
+        }
+        // TODO: call the autogenerated method to get the dynamic ids
+        // TODO: optimize caching
+        std::vector<int64_t> dynamicIds;
+        for (auto id : dynamicIds) {
+            std::pair<int64_t, int64_t> p{ep->getId(), id};
+            auto it = _entryPointCache.find(p);
+            if (it != _entryPointCache.end()) {
+                _allEntryPoints.push_back(it->second);
+            } else {
+                EntryPoint* nep = new EntryPoint(*ep);
+                nep->setDynamicId(id);
+                _entryPointCache.insert({p, nep});
+                _allEntryPoints.push_back(nep);
+            }
+        }
+    }
+    constexpr std::size_t MAX_CACHE_SIZE = 1000;
+    while (_entryPointCache.size() > MAX_CACHE_SIZE) {
+        _entryPointCache.erase(_entryPointCache.begin());
```

Comment:
This can remove valid recent entries + doesn't scale well. Once cache reaches 1000 entries, this will likely hit every time there are new dynamic id's.  On reaching max cache size, you can iterate over latest dynamic id's  and remove everything from cache that's not valid. + take max_cache_size from alica.yaml

Reply:
+ entryPoint is stored as const pointer, you will have to delete it first or better store as unique_ptr

Reply:
- yes, I have put a TODO to decide the cache policy
- Will store as unique_ptr

Reply:
implemented lru caching